src/EditorFeatures/CSharpTest2/Recommendations/WhenKeywordRecommenderTests.cs

```diff
@@ -9,67 +9,233 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Recommendations
+        public async Task TestForCatchClause_AfterCatch()
         {
             await VerifyKeywordAsync(AddInsideMethod(
 @"try {} catch $$"));
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
-        public async Task TestAfterCatchDeclaration1()
+        public async Task TestForCatchClause_AfterCatchDeclaration1()
         {
             await VerifyKeywordAsync(AddInsideMethod(
 @"try {} catch (Exception) $$"));
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
-        public async Task TestAfterCatchDeclaration2()
+        public async Task TestForCatchClause_AfterCatchDeclaration2()
         {
             await VerifyKeywordAsync(AddInsideMethod(
 @"try {} catch (Exception e) $$"));
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
-        public async Task TestAfterCatchDeclarationEmpty()
+        public async Task TestForCatchClause_AfterCatchDeclarationEmpty()
         {
             await VerifyKeywordAsync(AddInsideMethod(
 @"try {} catch () $$"));
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
-        public async Task TestNotAfterTryBlock()
+        public async Task TestForCatchClause_NotAfterTryBlock()
         {
             await VerifyAbsenceAsync(AddInsideMethod(
 @"try {} $$"));
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
-        public async Task TestNotAfterFilter1()
+        public async Task TestForCatchClause_NotAfterFilter1()
         {
             await VerifyAbsenceAsync(AddInsideMethod(
 @"try {} catch (Exception e) when $$"));
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
-        public async Task TestNotAfterFilter2()
+        public async Task TestForCatchClause_NotAfterFilter2()
         {
             await VerifyAbsenceAsync(AddInsideMethod(
 @"try {} catch (Exception e) when ($$"));
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
-        public async Task TestNotAfterFilter3()
+        public async Task TestForCatchClause_NotAfterFilter3()
         {
             await VerifyAbsenceAsync(AddInsideMethod(
 @"try {} catch (Exception e) when (true) $$"));
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
         [WorkItem(24113, "https://github.com/dotnet/roslyn/issues/24113")]
-        public async Task TestInCasePattern()
+        public async Task TestForSwitchCase_AfterDeclarationPattern()
         {
-            await VerifyKeywordAsync(AddInsideMethod(
-@"switch (1) { case int i $$ }"));
+            await VerifyKeywordAsync(AddInsideMethod(@"switch (1) { case int i $$ }"));
+            await VerifyKeywordAsync(AddInsideMethod(@"switch (1) { case int i $$: }"));
+            await VerifyKeywordAsync(AddInsideMethod(@"switch (1) { case int i $$ break; }"));
```

Comment:
note: in practice we prefer one test per test method.  That way if something fails, you don't have to fix it, only to find the next line then fails.

Reply:
hmm.. i actually thought about making a helper for this to test the same code all these cases... I think it would clutter up the tests to have 3 versions for each one. And when working on this, it never happened to me that 1 version would actually fail while another one didn't. I just added this as a precaution.

Reply:
VerifyKeywordAsync itself actually does like 5 different tests for variations such as a partially written keyword, and it's all in 1 method

Reply:
my point is... I don't think it's very likely that these would fail independently (as opposed to the tests hidden inside VerifyKeywordAsync, which often do)